Uplift/Views/Cells/FacilitiesDropdownCell.swift

```diff
@@ -9,136 +9,96 @@ import UIKit
+
+        headerView.delegate = self
 
-        setupDropdownView()
-        setupConstraints()
+        dropdownView = DropdownView(delegate: self,
+                                    headerView: headerView,
+                                    headerViewHeight: FacilitiesDropdownCell.headerViewHeight,
+                                    contentView: collectionView,
+                                    contentViewHeight: 0)
+        dropdownView.backgroundColor = .white
+        dropdownView.layer.masksToBounds = false
+        contentView.addSubview(dropdownView)
     }
 
     required init?(coder: NSCoder) {
         fatalError("init(coder:) has not been implemented")
     }
 
-    func configure(for facility: Facility) {
+    func configure(for facility: Facility, index: Int, dropdownStatus: DropdownStatus, headerViewTapped: @escaping (Int) -> ()) {
+        setNeedsUpdateConstraints()
         self.facility = facility
+        self.facilitiesIndex = index
+        self.headerViewTapped = headerViewTapped
+        dropdownView.updateContentViewHeight(to: FacilitiesDropdownCell.getHeights(for: facility) - FacilitiesDropdownCell.headerViewHeight)
+        headerView.configure(for: facility)
+        if dropdownStatus == .open {
+            self.dropdownView.openDropdown()
+        } else {
+            self.dropdownView.closeDropdown()
+        }
+            self.dropdownView.layoutIfNeeded()
         collectionView.reloadData()
     }
 
-    func setupDropdownView() {
-        let headerViewHeight: CGFloat = 52
-        setupHeaderView(headerString: facility.name)
-
-        // TODO: register facility cell classes once they're finished
-
-        collectionViewHeight = 0 // TODO: once cell classes are finished, sum up their heights to get the total collection view height
-
-        dropdownView = DropdownView(delegate: self,
-                                    headerView: headerView,
-                                    headerViewHeight: headerViewHeight,
-                                    contentView: collectionView,
-                                    contentViewHeight: collectionViewHeight)
-    }
-
-    func setupHeaderView(headerString: String) {
-        var headerImage: UIImage?
-        switch FacilityName(rawValue: headerString.uppercased()) {
-        case .equipment, .fitnessCenter:
-            headerImage = UIImage(named: ImageNames.equipment)
-        case .gymnasium:
-            headerImage = UIImage(named: ImageNames.basketball)
-        case .swimmingPool:
-            headerImage = UIImage(named: ImageNames.pool)
-        case .bowlingLanes:
-            headerImage = UIImage(named: ImageNames.bowling)
-        default:
-            headerImage = UIImage(named: ImageNames.misc)
+    override func updateConstraints() {
+        super.updateConstraints()
+        dropdownView.snp.makeConstraints { make in
+            make.leading.trailing.top.bottom.equalToSuperview()
```

Comment:
why are we using `updateConstraints`? we shouldn't need it. you can do `updateConstraints/remakeConstraints` if you need to change constraints

Reply:
bump on this